### ISSUE TITLE
Which repositories cost us a lot

### ISSUE THREAD

**Issue by betatim** [jupyterhub/mybinder.org-deploy#643]
as a continuation of #628:

> In another issue we should discuss how to get an insight into which repositories are costing a lot of money.

There are at least two things here: how to define "cost" and how to obtain the metrics.

It would be good to know for example if the jupyter.org/try launches are costing us a lot of money or not. For example: Are most of the compute resources used by these launches (they are about 2/3 of all launches) or are they essentially free because people don't compute a lot on them?

**Comment by betatim** [comment]
What I was thinking of is a repo that gets launched but people just briefly look at something and then navigate away vs a repo people launch to then train a deep learning model for 5h. The point is that just because a repository is popular it doesn't mean it costs a lot because each session is super short.

A number I'd try to get to help look at this is "minutes live on mybinder": the total number of minutes instances for a repo were live on mybinder.org for the day/week/month.

**Comment by choldgraf** [comment]
yeah that's a good point - I like the "minutes live on mybinder" idea

**Comment by betatim** [comment]
What do people think of using the start and stop life cycle methods to trigger a HTTP request to a API that logs when a pod starts and stops and then does the calculation of "live minutes per repo"? One thing to check is if the lifecycle methods get called when a pod gets (forcefully) deleted.

I am surprised that I can't find anything like this ready made for k8s :-/

**Comment by choldgraf** [comment]
well I do believe that we can get pod age information out of prometheus, we could crawl back through history that way, no?

**Comment by betatim** [comment]
`container_cpu_cfs_throttled_seconds_total{pod_name=~"^jupyter-.*"}` parking this query here. I think this might lead somewhere interesting along what you suggested Chris. Can't quite work out how to group/sum this by pod name minus "the end", in order to group pods from the same repository together.
